azimuth/modules/dataset_analysis/similarity_analysis.py

```diff
@@ -73,30 +73,28 @@ class NeighborsTaggingModule(DatasetResultModule[SimilarityConfig]):
     """Compute neighbors for each utterance in a dataset split."""
 
     def compute_on_dataset_split(self) -> List[TaggingResponse]:  # type: ignore
-        if DatasetSplitName.train in self.available_dataset_splits:
-            train_dm: Optional[DatasetSplitManager] = self.get_dataset_split_manager(
-                DatasetSplitName.train
-            )
-            # NOTE: Doing this ensure that the FAISS index exists somewhere.
-            # If FAISSModule fails this module will fail instead of blocking.
-            train_features: Optional[List[Array]] = self._get_features(DatasetSplitName.train)
-        else:
-            train_dm = None
-            train_features = None
-
-        eval_dm = self.get_dataset_split_manager(DatasetSplitName.eval)
-        eval_features = self._get_features(DatasetSplitName.eval)
-        ds_features = assert_not_none(
+        def get_dm_and_features(ds_name):
+            if ds_name in self.available_dataset_splits:
+                dm: Optional[DatasetSplitManager] = self.get_dataset_split_manager(ds_name)
+                features: Optional[List[Array]] = self._get_features(ds_name)
+                return dm, features
+            else:
+                return None, None
+
+        train_dm, train_features = get_dm_and_features(DatasetSplitName.train)
+        eval_dm, eval_features = get_dm_and_features(DatasetSplitName.eval)
+
+        ds_features: List[Array] = assert_not_none(
             eval_features if self.dataset_split_name == DatasetSplitName.eval else train_features
         )
+        ds_dm: DatasetSplitManager = assert_not_none(
+            eval_dm if self.dataset_split_name == DatasetSplitName.eval else train_dm
+        )
```

Comment:
That looks equivalent to something like this, considering `get_dataset_split_manager` already asserts that the dataset split is not `None`. Am I missing something?
```suggestion
        dm = self.get_dataset_split_manager(self.dataset_split_name)
        features = self._get_features(self.dataset_split_name)
```
I'm not sure if we need to wrap `features = assert_not_none(...)`.

Reply:
Thank you for pointing that out. My understanding and changes were completely inadequate. Turns out that the comment was useful. I rewrote it so it's clearer. Let me know what you think.

Reply:
Thank you for taking the time to dig into this and to make sure we understand what is going on. Good job!